presto-spi/src/main/java/com/facebook/presto/spi/analyzer/QueryAnalysis.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.analyzer;
+
+import com.facebook.presto.spi.function.FunctionKind;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * The QueryAnalysis interface should be used to represent the query semantic analysis.
+ * This interface should be implemented by respective analyzer to represent query semantic analysis.
+ */
+public interface QueryAnalysis
```

Comment:
should we use the opportunity and rename the interface to something better? not sure I have a better suggestion without going to something like BoundAST/LogicalQueryTree etc

Reply:
I am happy to rename interfaces/methods to anything better. I would say lets discuss this offline and see if there are better names available. We can always make those changes.

Reply:
one option would be AnalyzerResult